presto-main/src/test/java/io/prestosql/sql/planner/TestDynamicFilter.java

```diff
@@ -119,6 +119,33 @@ public void testJoinOnCast()
                                                 tableScan("orders", ImmutableMap.of("ORDERS_OK", "orderkey")))),
                                 anyTree(
                                         tableScan("lineitem", ImmutableMap.of("LINEITEM_OK", "orderkey"))))));
+
+        assertPlan("SELECT o.orderkey FROM orders o, lineitem l WHERE l.orderkey = cast(o.orderkey as int)",
```

Comment:
what is the check here? That DF exist or not? Can' tell from the test (and the one above too)

Reply:
Made the test more clear, DF should not exist in these cases

Reply:
> Made the test more clear, DF should not exist in these cases

`anyTree` will match `filterNode` that could have DFs. I don't think this `join` matcher ensures that there is no DF, does it?

Reply:
yes that's right, I missed it
i've updated the tests to avoid anyTree and explicitly match no DF now.